server/src/main/java/org/candlepin/pinsetter/tasks/KingpinJob.java

```diff
@@ -131,23 +130,40 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
          *  from specific JDBC drivers which wrap SQLException. We let the jobs themselves
          *  sort this out and throw a specific exception to indicate a retry is an option.
          */
-        catch (PersistenceException e) {
-            if (!context.getJobDetail().getKey().getName().startsWith("hypervisor")) {
-                refireCheck(context, e);
-            }
-            else {
+        catch (LockTimeoutException | OptimisticLockException | PessimisticLockException e) {
+            dropQueuedEvents();
+            if (isHypervisor(context)) {
                 log.debug("Hypervisor job failure");
                 throw new JobExecutionException(e, false);
             }
-            if (eventSink != null) {
-                eventSink.rollback();
-            }
-        }
-        catch (RetryJobException e) {
             refireCheck(context, e);
-            if (eventSink != null) {
-                eventSink.rollback();
+        }
+        catch (RuntimeException e) {
+            dropQueuedEvents();
+            Throwable cause = e.getCause();
+            while (cause != null) {
+                if (SQLException.class.isAssignableFrom(cause.getClass())) {
+                    log.warn("Caught a runtime exception wrapping an SQLException.");
+                    refireCheck(context, e);
+                }
+                cause = cause.getCause();
             }
+            // Otherwise throw as we would normally for any generic Exception:
+            log.error("RefreshPoolsJob encountered a problem.", e);
```

Comment:
Now that this code lives in KingpinJob, the logging should change to output the subclass' name here, using `this.getClass().getName()`.